tasks/post-to-cmr/index.js

```diff
@@ -167,9 +183,124 @@ function buildOutput(results, input, granules, regex) {
+* @param {string} sourceBucket - source bucket location of files
+* @returns {Promise} promise resolves when all files have been moved
+**/
+async function moveGranuleFiles(granulesObject, sourceBucket) {
+  const moveFileRequests = [];
+
+  Object.keys(granulesObject).forEach((granuleKey) => {
+    const granule = granulesObject[granuleKey];
+    granule.files.forEach((file) => {
+      const source = {
+        Bucket: sourceBucket,
+        Key: `${file.fileStagingDir}/${file.name}`
+      };
+
+      const target = {
+        Bucket: file.bucket,
+        Key: file.filepath
+      };
+
+      moveFileRequests.push(moveGranuleFile(source, target));
+    });
+  });
+
+  return Promise.all(moveFileRequests);
 }
 
+/**
+* Update the granule metadata with the final location of files
+*
+* @param {Object} granulesObject - an object of the granules where the key is the granuleId
+* @param {Object} collection - configuration object defining a collection
+* of granules and their files
+* @param {string} cmrFiles - array of objects that include CMR xmls uris and granuleIds
+* @returns {Promise} promise resolves when all files have been moved
+**/
+function updateGranuleMetadata(granulesObject, collection, cmrFiles) {
+  Object.keys(granulesObject).forEach((granuleId) => {
+    granulesObject[granuleId].files.forEach((file) => {
+      collection.files.forEach((fileConfig) => {
+        const test = new RegExp(fileConfig.regex);
+        const match = file.name.match(test);
+
+        const cmrFile = cmrFiles.find((f) => f.granuleId === granuleId);
+
+        if (match) {
+          if (!file.url_path) {
+            file.url_path = collection.url_path || '';
+          }
+
+          const urlPath = urlPathTemplate(file.url_path, {
+            file: file,
+            granule: granulesObject[granuleId],
+            cmrMetadata: cmrFile.metadataObject
+          });
+
+          file.bucket = fileConfig.bucket;
+          file.filepath = path.join(urlPath, file.name);
+          file.filename = `s3://${path.join(file.bucket, file.filepath)}`;
+        }
+      });
+    });
+  });
+
+  return granulesObject;
+}
+
+/**
+* Update the online access url fields in CMR xml files
+*
+* @param {string} cmrFiles - array of objects that include CMR xmls uris and granuleIds
+* @param {Object} granulesObject - an object of the granules where the key is the granuleId
+* @returns {Promise} promise resolves when all files have been updated
+**/
+async function updateCmrFileAccessURls(cmrFiles, granulesObject) {
+  for (const cmrFile of cmrFiles) {
+    const onlineAccessUrls = cmrFile.metadataObject.Granule.OnlineAccessURLs;
+
+    const urls = onlineAccessUrls.OnlineAccessURL.map((urlObj) => {
+      const filename = path.basename(urlObj.URL);
+      const granule = granulesObject[cmrFile.granuleId];
+      const file = granule.files.find((f) => f.name === filename);
+      urlObj.URL = file.filename;
```

Comment:
For the files in the protected bucket, the url starts with the distribution api:
https://3tu0izwxuc.execute-api.us-east1.amazonaws.com/dev/MOD09GQ.A2016358.h13v04.006.2016360104606.hdf
For the files in public bucket, the url could be s3 bucket/key:
http://cumulus-test-sandbox-public.s3.amazonaws.com/MOD09GQ.A2016358.h13v04.006.2016360104606_ndvi.jpg
Are we replacing everything with s3://?  Is it a valid link for user to download?

Reply:
You're right this should be revised. Working on that now.

Might be good to merge #325 first as it has some buckets-related changes that would be useful here. For now I think I'll branch off of Cumulus-456-integration-test.

Reply:
Seth branch off Cumulus-456-integration-test-aws-test if you're branching